google-cloud-spanner-cassandra/src/main/java/com/google/cloud/spanner/adapter/DriverConnectionHandler.java

```diff
@@ -248,14 +261,19 @@ private PreparePayloadResult preparePayload(byte[] payload) {
     payloadBuf.release();
 
     Map<String, String> attachments = new HashMap<>();
+    if ((frame.message instanceof Execute || frame.message instanceof Batch)
+        && maxCommitDelayMillis.isPresent()) {
+      attachments.put(MAX_COMMIT_DELAY_ATTACHMENT_KEY, maxCommitDelayMillis.get());
+    }
+
```

Comment:
This should only be added for write operations, similar to LAR feature

Reply:
fixed